app/scripts/metamask-controller.js

```diff
@@ -4784,8 +4784,27 @@ export default class MetamaskController extends EventEmitter {
    * @param {string} [options.subjectType] - The type of the sender, i.e. subject.
    */
   setupUntrustedCommunication({ connectionStream, sender, subjectType }) {
-    const { completedOnboarding } = this.onboardingController.store.getState();
-    const { usePhishDetect } = this.preferencesController.store.getState();
+    if (sender.url) {
+      if (this.onboardingController.store.getState().completedOnboarding) {
+        if (this.preferencesController.store.getState().usePhishDetect) {
```

Comment:
This logic changed at one point, refactored, and finally restored to the original logic, but I kept some of the refactoring.

All the refactoring does is minimize the amount of "work" done in these pre-condition checks.

I don't mind restoring it to the original, as the refactoring's performance improvements here are certainly insignificant.

Reply:
this refactor looks fine